conductr_cli/bndl_main.py

```diff
@@ -83,18 +95,70 @@ def build_parser():
+                        nargs='?',
+                        required=False,
+                        help='Sets the "compatibilityVersion" bundle.conf value',
+                        dest='compatibility_version')
 
-    for argument, bundle_key in mappings.items():
-        help_text = 'Sets the "{}" bundle.conf value'.format(bundle_key)
+    parser.add_argument('--disk-space',
+                        nargs='?',
+                        required=False,
+                        help='Sets the "diskSpace" bundle.conf value',
+                        dest='disk_space',
+                        type=int)
+
+    parser.add_argument('--memory',
+                        nargs='?',
+                        required=False,
+                        help='Sets the "memory" bundle.conf value',
+                        dest='memory',
+                        type=int)
 
-        parser.add_argument(argument,
-                            nargs='*' if argument in zero_or_more_mappings else '?',
-                            type=type_mappings[argument] if argument in type_mappings else None,
-                            required=False,
-                            help=help_text,
-                            dest=bundle_key)
+    parser.add_argument('--name',
+                        nargs='?',
+                        required=False,
+                        help='Sets the "name" bundle.conf value',
+                        dest='name')
+
+    parser.add_argument('--nr-of-cpus',
+                        nargs='?',
+                        required=False,
+                        help='Sets the "nrOfCpus" bundle.conf value',
+                        dest='nr_of_cpus',
+                        type=float)
+
+    parser.add_argument('--roles',
+                        nargs='*',
+                        required=False,
+                        help='Sets the "roles" bundle.conf value',
+                        dest='roles')
+
+    parser.add_argument('--system',
+                        nargs='?',
+                        required=False,
+                        help='Sets the "system" bundle.conf value',
+                        dest='system')
+
+    parser.add_argument('--system-version',
+                        nargs='?',
+                        required=False,
+                        help='Sets the "systemVersion" bundle.conf value',
+                        dest='system_version')
+
+    parser.add_argument('--tag',
+                        action='append',
+                        default=[],
+                        dest='tags',
+                        help='Tags to add to bundle.conf\n'
+                             'Example: bndl --tag 16.04 --tag xenial\n'
+                             'Defaults to []')
+
+    parser.add_argument('--version',
+                        nargs='?',
+                        required=False,
+                        help='Sets the "version" bundle.conf value',
+                        dest='version')
 
```

Comment:
Refactored into explicit listing of arguments. A bit repetitive but it simplified the code.